runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -146,9 +146,21 @@ public VersionLockedObject(Supplier<T> newObjectFn,
         if (ts != 0) {
             if (directAccessCheckFunction.apply(this)) {
                 log.trace("Access [{}] Direct (optimistic-read) access at {}", this, getVersionUnsafe());
-                R ret = accessFunction.apply(object);
-                if (lock.validate(ts)) {
-                    return ret;
+                try {
```

Comment:
Well, I could add a flag to switch between optimistic and full blown read locks and we could benchmark. When there's no contention, the optimistic read is definitely much faster than the full read lock, so it's definitely very workload dependent.